app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt

```diff
@@ -115,6 +115,7 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
             input.doOnTextChanged(onSixDigitsHandler(digits, requireContext()) { digitsAsString ->
                 if (passcodeArg == digitsAsString) {
                     viewModel.setupPasscode(digitsAsString)
+                    viewModel.encryptPasscodeWithBiometricKey(digitsAsString)
```

Comment:
This is the fix. Update after setting the passcode

Reply:
what happens if device doesn't support biometrics?

Reply:
This does not depend on the biometrics capability. But OTOH it is not necessary on devices that do not have biometrics functionality at all.

Reply:
so check should be added to see if this action is necessary? Also this seems to be part of password setup. Why not combine it in a view model?

Reply:
please add todo's, fixme's and let's revisit it later